include/oneapi/mkl/dft/descriptor.hpp

```diff
@@ -51,22 +58,28 @@ class descriptor {
 
     void get_value(config_param param, ...);
 
-    void commit(sycl::queue& queue);
+    void commit(sycl::queue& queue) {
+        pimpl_.reset(detail::create_commit(get_device_id(queue), queue, values));
+    }
 
 #ifdef ENABLE_MKLCPU_BACKEND
-    void commit(backend_selector<backend::mklcpu> selector);
+    void commit(backend_selector<backend::mklcpu> selector) {
+        pimpl_.reset(mklcpu::create_commit(selector.get_queue(), values));
+    }
 #endif
 
 #ifdef ENABLE_MKLGPU_BACKEND
-    void commit(backend_selector<backend::mklgpu> selector);
+    void commit(backend_selector<backend::mklgpu> selector) {
```

Comment:
Should these `commit(backend_selector<BACKEND>)` functions also accept a queue as an argument? I might have multiple GPUs in a system, so the selector might not have  a 1-to-1 mapping of backends to queues.

Reply:
I Imagine we have mklgpu, nvidia support for now, with that there should be a 1-to-1 mapping of backends_selector with the queue.

Reply:
Ah, I see that the `backend_selector` is constructed with a specific queue. My mistake.

Instead I should have asked about `void commit(sycl::queue& queue)` where a queue might not have a 1-1 mapping with a backend. Eg. The CPU queue might support both MKLCPU and FFTW, and a IntelGPU queue might support MLKGPU and VkFFT's level zero backend.

Reply:
@hjabird do I understand correctly that this comment is to figure out how to support multiple backends for the same device ie how void commit(sycl::queue& queue) would decide to dispatch between the oneMKL CPU wrapper or the FFTW CPU wrapper for example?

Reply:
So if my understanding is correct, the current approach in the project is that the API (e.g. BLAS API) takes only the sycl queue. If several backends are available for the same device, they can't be used simultaneously in the same app. We have to chose a priority order for each backends for each device. When the library is initialized it will check which third party library is available and set for each device the available backend that have the highest priority.
For example for DFT if both MKLCPU and FFTW backends are available and MKLCPU is chosen to have higher priority then MKLCPU will be used. But if only FFTW library is found then FFTW backend will be used.

You can see an example for BLAS in function_table_intializer.hpp lines 95-100 and backends_table.hpp around line 50.

So the commit API does not need to change but the header files mentioned above need to be updated with the DFT supported backends and the appropriate function tables for each backend need to be created for DFT.